maps/randomvaults/objects.dm

```diff
@@ -649,3 +657,207 @@
 
 		CO.face_atom(AM)
 		CO.visible_message("<span class='notice'>\The [CO] looks at \the [AM] and hisses angrily!</span>")
+
+
+/* Spy Sat stuff */
+
+/obj/machinery/power/magtape_deck
+	name = "magnetic tape drive"
+	desc = "A primitive way of storing information. Used because of its longevity over most digital counterparts."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "blackbox"
+
+	active_power_usage = 500
+	density = 1
+	var/powered
+
+/obj/machinery/power/magtape_deck/New()
+	..()
+	connect_to_network()
+
+/obj/machinery/power/magtape_deck/process()
+	if(stat & BROKEN)
+		return
+	if(avail(active_power_usage))
+		powered = 1
+	else
+		powered = 0
+
+	if(powered && stat & NOPOWER)
+		stat &= ~NOPOWER
+		update_icon()
+	else if (!powered && !(stat & NOPOWER))
+		stat |= NOPOWER
+		update_icon()
+
+/obj/machinery/power/magtape_deck/update_icon()
+	if(stat & (BROKEN|NOPOWER))
+		icon_state = "[initial(icon_state)]0"
+	else
+		icon_state = "[initial(icon_state)]1"
+
+/obj/machinery/power/magtape_deck/emp_act(severity)
+	if(prob(50/severity))
+		set_broken()
+
+/obj/machinery/power/magtape_deck/ex_act(severity)
+	switch(severity)
+		if(1)
+			qdel(src)
+		if(2)
+			if (prob(50))
+				set_broken()
+		if(3)
+			if (prob(25))
+				set_broken()
+
+/obj/machinery/power/magtape_deck/proc/set_broken()
+	if(stat & BROKEN)
+		return
+	stat |= BROKEN
+	update_icon()
+
+/obj/machinery/power/magtape_deck/blob_act()
+	if (prob(75))
+		set_broken()
+
+
+/obj/machinery/power/magtape_deck/syndicate
+	desc = "A primitive way of storing information. Used because of its longevity over most digital counterparts.\
+	It appears to have a microphone and speaker attached."
+	var/codephrase
+	var/encrypted_codephrase
+	var/triggered
+	flags = FPRINT | HEAR
+
+/obj/machinery/power/magtape_deck/syndicate/New()
+	..()
+	generate_codephrase()
+
+/obj/machinery/power/magtape_deck/syndicate/proc/generate_codephrase()
+	triggered = 0
+	codephrase = pick(adjectives)
+	var/offset = rand(1,5)
+	for(var/i = 1;i < length(codephrase)+1; i++)
```

Comment:
Please comment these lines. It's extremely unreadable.

Also, use `for (var/i = 1 to length(codephrase))`